DealMart/products/views/products.py

```diff
@@ -16,17 +17,34 @@ def get(self,request):
 
 
     def post(self,request):
+        # import pdb;pdb.set_trace()
+        # print(request.FILES)
         form  = AddProductForm(request.POST,request.FILES)
+        
+        # print(images)
 
-        if form.is_valid():
-            name = form.cleaned_data.get('name')
-            category = form.cleaned_data.get('category')
-            sub_category = form.cleaned_data.get('sub_category')
-            price = form.cleaned_data.get('price')
-            image = form.cleaned_data.get('image')
-            discription  = form.cleaned_data.get('discription')
+
+        if form.is_valid(): 
+            images = request.FILES['image']
+            if images == None:
+                images="static/images/default.jpg"
+            name = request.POST.get('name')
+            category_id = request.POST.get('category')
+            category = Category.objects.get(id=category_id)
+            sub_category_name = request.POST.get('sub_category')
+            sub_category = SubCategory.objects.get(name=sub_category_name)
```

Comment:
Use id to get product instead of name because if two sub-category are having same name then it will throw an error.